imag-todo/src/main.rs

```diff
@@ -16,210 +16,144 @@ extern crate libimagtodo;
-                            args.push(format!("uuid:{}", uuid));
-                            args.push(format!("{}", "information"));
-                            let tw_process = Command::new("task").stdin(Stdio::null()).args(&args).spawn()
-                                .unwrap_or_else(|e| {
-                                    error!("{}", e);
-                                    panic!("failed");
-                                });
-                            let output = tw_process.wait_with_output().unwrap_or_else(|e| {
-                                panic!("failed to unwrap output: {}", e);
-                            });
-                            let outstring = String::from_utf8(output.stdout).unwrap_or_else(|e| {
-                                panic!("failed to ececute: {}", e);
-                            });
-                            println!("{}", outstring);
-                        }
-                        else {
-                            println!("{}", match uuid {
-                                toml::Value::String(s) => s,
-                                _ => {
-                                    error!("Unexpected type for todo.uuid: {}", uuid);
-                                    continue;
-                                },
-                            });
                         }
                     }
+                } // end if c % 2
+            },
+            Err(e) => {
+                trace_error(&e);
+                exit(1);
+            },
+        }
+    } else {
+        // Should not be possible, as one argument is required via
+        // ArgGroup
+        unreachable!();
+    }
+}
+
+fn list(rt: &Runtime) {
+    let subcmd  = rt.cli().subcommand_matches("list").unwrap();
+    let verbose = subcmd.is_present("verbose");
+
+    let res = Task::all(rt.store())
+        .map(|iter| {
+            let uuids : Vec<_> = iter.filter_map(|t| match t {
+                Ok(v) => match v.get_header().read("todo.uuid") {
+                    Ok(Some(Value::String(ref u))) => Some(u.clone()),
+                    Ok(Some(_)) => {
+                        warn!("Header type error");
+                        None
+                    },
+                    Ok(None) => None,
                     Err(e) => {
-                        error!("{}", e);
-                        continue;
+                        trace_error(&e);
+                        None
                     }
-                } // end match task
-            } // end for
-        }
-        _ => unimplemented!(),
-    } // end match scmd
-} // end main
+                },
+                Err(e) => {
+                    trace_error(&e);
+                    None
+                }
+            })
+            .collect();
+
+            let outstring = if verbose {
+                let output = Command::new("task")
+                    .stdin(Stdio::null())
+                    .args(&uuids)
```

Comment:
Does this work that way? I know that you can filter tasks in tw, but I'm not sure if you can just put in all UUID's. If you tested it, and it works, then nevermind. If it doesnt, iterate over the uuids and call tw for each of it. I kinda hope it works the way you have written it.
Also, there are several levels of verbosity. Your way, tw behaves as if the command it gets was 'next'. But there's also the command 'information', which gives very detailed information about tasks.
So it is possibility to allow multiple occurences of --verbose (-v) and introduce verbosity levels for this. Though it isn't really necessary for now (I think).